src/SharpCompress/Archives/IWritableArchiveExtensions.cs

```diff
@@ -39,15 +39,21 @@ public static void SaveTo(this IWritableArchive writableArchive, FileInfo fileIn
             this IWritableArchive writableArchive,
             string filePath, string searchPattern = "*.*", SearchOption searchOption = SearchOption.AllDirectories)
         {
+            using (writableArchive.PauseEntryRebuilding())
+            {
 #if NET35
-            foreach (var path in Directory.GetFiles(filePath, searchPattern, searchOption))
+                foreach (var path in Directory.GetFiles(filePath, searchPattern, searchOption))
 #else
-            foreach (var path in Directory.EnumerateFiles(filePath, searchPattern, searchOption))
+                foreach (var path in Directory.EnumerateFiles(filePath, searchPattern, searchOption))
 #endif
-            {
-                var fileInfo = new FileInfo(path);
-                writableArchive.AddEntry(path.Substring(filePath.Length), fileInfo.OpenRead(), true, fileInfo.Length,
-                                         fileInfo.LastWriteTime);
+                {
+                    var fileInfo = new FileInfo(path);
+                    writableArchive.AddEntry(path.Substring(filePath.Length),
+                                             fileInfo.OpenRead(),
+                                             true,
+                                             fileInfo.Length,
+                                             fileInfo.LastWriteTime);
+                }
```

Comment:
How about instead of using IDispose we have:
void PauseEntryBuilding(Action<.....> action)
{
 Pause();
 try
 {
   action(...);
 }
 finally
 {
   Resume();
 }
}

called like 

PauseEntryBuilding((....) => 
{
  foreach .... addEntry....
});

Reply:
That depends on which version of C# you use - this is a bit functional programming like, but completely normal.